greenmail-core/src/test/java/com/icegreen/greenmail/test/SendReceiveWithInternationalAddress.java

```diff
@@ -0,0 +1,167 @@
+
+            if (type == RecipientType.NEWSGROUPS) {
+
+                if (addresses == null || addresses.length() == 0)
+                    removeHeader("Newsgroups");
+                else
+                    setHeader("Newsgroups", addresses);
+
+            } else {
+
+                if (!isStrict()) {
+
+                    InternetAddress address = new InternetAddress();
+                    address.setAddress(addresses);
+                    setAddressHeader(getHeaderName(type), new InternetAddress[]{address});
+
+                } else {
+
+                    setAddressHeader(getHeaderName(type),
+
+                            addresses == null ? null : InternetAddress.parse(addresses));
+                }
+            }
+        }
+
+        // Current Java Mail version does not respect the "mail.mime.address.strict"
+        // value when setting email addresses in a MimeMessage object (calls
+        // InternetAddress.parse(String), which will always validate the
+        // address based on rfc822) so we need to override this method to
+        // make this example work.
+        @Override
+        public void setFrom(String address) throws MessagingException {
+
+            if (address == null)
+                    removeHeader("From");
+
+            else {
+
+                if (!isStrict()) {
+
+                    InternetAddress internetAddress = new InternetAddress();
+                    internetAddress.setAddress(address);
+                    setAddressHeader("From", new InternetAddress[]{internetAddress});
+
+                } else {
+
+                    setAddressHeader("From", InternetAddress.parse(address));
+                }
+            }
+        }
+
+        private boolean isStrict() {
+            return PropUtil.getBooleanSessionProperty(session, "mail.mime.address.strict", true);
+        }
+
+        // Convenience method to set addresses
+        private void setAddressHeader(String name, InternetAddress[] addresses)
+                throws MessagingException {
+
+            try {
+
+                // Encoding the email addresses so they are correctly sent to the mail server.
+                for (int i = 0; i < addresses.length; i++) {
+                    String addStr = MimeUtility.encodeText(addresses[i].getAddress());
+                    InternetAddress ia = new InternetAddress();
+                    ia.setAddress(addStr);
+                    addresses[i] = ia;
+                }
+
+                String s = InternetAddress.toString(addresses, name.length() + 2);
+                if (s == null)
+                    removeHeader(name);
+                else
+                    setHeader(name, s);
+            } catch (UnsupportedEncodingException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        private String getHeaderName(Message.RecipientType type)
+                throws MessagingException {
+            String headerName;
```

Comment:
Yes, because it is called from the `setRecipients` method that we are already overriding so we would need this or to implement this code on that method. I'm removing an unused `else if` block.